hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveShimLoader.java

```diff
@@ -0,0 +1,149 @@
+  public static final String HIVE_VERSION_V1_2_0 = "1.2.0";
+  public static final String HIVE_VERSION_V1_2_1 = "1.2.1";
+  public static final String HIVE_VERSION_V1_2_2 = "1.2.2";
+  public static final String HIVE_VERSION_V2_0_0 = "2.0.0";
+  public static final String HIVE_VERSION_V2_0_1 = "2.0.1";
+  public static final String HIVE_VERSION_V2_1_0 = "2.1.0";
+  public static final String HIVE_VERSION_V2_1_1 = "2.1.1";
+  public static final String HIVE_VERSION_V2_2_0 = "2.2.0";
+  public static final String HIVE_VERSION_V2_3_0 = "2.3.0";
+  public static final String HIVE_VERSION_V2_3_1 = "2.3.1";
+  public static final String HIVE_VERSION_V2_3_2 = "2.3.2";
+  public static final String HIVE_VERSION_V2_3_3 = "2.3.3";
+  public static final String HIVE_VERSION_V2_3_4 = "2.3.4";
+  public static final String HIVE_VERSION_V2_3_5 = "2.3.5";
+  public static final String HIVE_VERSION_V2_3_6 = "2.3.6";
+
+  private static final Map<String, HiveShim> HIVE_SHIMS = new ConcurrentHashMap<>(2);
+
+  private HiveShimLoader() {
+  }
+
+  public static HiveShim loadHiveShim(String version) {
+    return HIVE_SHIMS.computeIfAbsent(version, (v) -> {
+      if (v.startsWith(HIVE_VERSION_V1_0_0)) {
+        return new HiveShimV100();
+      }
+      if (v.startsWith(HIVE_VERSION_V1_0_1)) {
+        return new HiveShimV101();
+      }
+      if (v.startsWith(HIVE_VERSION_V1_1_0)) {
+        return new HiveShimV110();
+      }
+      if (v.startsWith(HIVE_VERSION_V1_1_1)) {
+        return new HiveShimV111();
+      }
+      if (v.startsWith(HIVE_VERSION_V1_2_0)) {
+        return new HiveShimV120();
+      }
+      if (v.startsWith(HIVE_VERSION_V1_2_1)) {
+        return new HiveShimV121();
+      }
+      if (v.startsWith(HIVE_VERSION_V1_2_2)) {
+        return new HiveShimV122();
+      }
+      if (v.startsWith(HIVE_VERSION_V2_0_0)) {
+        return new HiveShimV200();
+      }
+      if (v.startsWith(HIVE_VERSION_V2_0_1)) {
+        return new HiveShimV201();
+      }
+      if (v.startsWith(HIVE_VERSION_V2_1_0)) {
+        return new HiveShimV210();
+      }
+      if (v.startsWith(HIVE_VERSION_V2_1_1)) {
+        return new HiveShimV211();
+      }
+      if (v.startsWith(HIVE_VERSION_V2_2_0)) {
+        return new HiveShimV220();
+      }
+      if (v.startsWith(HIVE_VERSION_V2_3_0)) {
+        return new HiveShimV230();
+      }
+      if (v.startsWith(HIVE_VERSION_V2_3_1)) {
+        return new HiveShimV231();
+      }
+      if (v.startsWith(HIVE_VERSION_V2_3_2)) {
+        return new HiveShimV232();
+      }
+      if (v.startsWith(HIVE_VERSION_V2_3_3)) {
+        return new HiveShimV233();
+      }
+      if (v.startsWith(HIVE_VERSION_V2_3_4)) {
+        return new HiveShimV234();
+      }
+      if (v.startsWith(HIVE_VERSION_V2_3_5)) {
+        return new HiveShimV235();
+      }
+      if (v.startsWith(HIVE_VERSION_V2_3_6)) {
+        return new HiveShimV236();
+      }
+      throw new HoodieHiveSyncException("Unsupported Hive version " + v);
```

Comment:
can you help me understand what do we do today if someone tries with higher hive versions? do we explicitly fail the call somewhere? Or are we changing the behavior with this patch ?